src/text-input/index.ts

```diff
@@ -152,27 +165,54 @@ export class TextInputBase<P extends TextInputProperties = TextInputProperties>
 		this.properties.onTouchCancel && this.properties.onTouchCancel();
 	}
 
-	private _uuid: string;
+	private _validate() {
+		const { validate, onValidate, value } = this.properties;
+		if (!validate || value === undefined || value === null) {
+			return;
+		}
+
+		let { valid, message } = this.meta(ValidityMeta).get('input', value);
 
-	constructor() {
-		super();
-		this._uuid = uuid();
+		if (typeof validate === 'function') {
+			const { valid: customValid, message: customMessage } = validate(value);
```

Comment:
do these need typing here given that `validate` is typed?

Reply:
Well, `validate` could be a boolean, a function, or `undefined`, so I need to type guard it and treat it and call it if it is a function.